From fef9381530bdeda69c0f4c0b95e65b12c6180392 Mon Sep 17 00:00:00 2001
From: The-going <48602507+The-going@users.noreply.github.com>
Date: Thu, 24 Mar 2022 17:21:08 +0300
Subject: [PATCH] Fix duplicate nodes for sun50i-h5-orangepi-pc2

The problem with duplicate nodes in the device tree is that the first one
in the queue will be applied. The subsequent one will be ignored, not replaced.

Fix spi-nor node initialization error for orangepi-pc2

[    1.288263] spi-nor spi0.0: supply vdd not found, using dummy regulator
[    1.306391] spi-nor spi0.0: mx25l1606e (2048 Kbytes)
[    1.845203] Freeing initrd memory: 10824K
[    1.855263] sun6i-spi 1c68000.spi: chipselect 0 already in use
[    1.861159] spi_master spi0: spi_device register error /soc/spi@1c68000/spi-flash@0
[    1.868842] spi_master spi0: Failed to create SPI device for /soc/spi@1c68000/spi-flash@0

Remove duplicate nodes from applied patches for the board
sun50i-h5-orangepi-pc2. After this fix:

[    1.290252] sun50i-h5-pinctrl 1c20800.pinctrl: supply vcc-pc not found, using dummy regulator
[    1.291250] spi-nor spi0.0: supply vdd not found, using dummy regulator
[    1.310397] spi-nor spi0.0: mx25l1606e (2048 Kbytes)
[    1.845686] Freeing initrd memory: 10824K
[    1.854333] 2 fixed-partitions partitions found on MTD device spi0.0
[    1.854369] Creating 2 MTD partitions on "spi0.0":
[    1.854381] 0x000000000000-0x000000100000 : "uboot"
[    1.855679] 0x000000100000-0x000000200000 : "env"
---
 .../dts/allwinner/sun50i-h5-orangepi-pc2.dts  | 32 -------------------
 1 file changed, 32 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h5-orangepi-pc2.dts b/arch/arm64/boot/dts/allwinner/sun50i-h5-orangepi-pc2.dts
index 181a465fb..0c821deec 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h5-orangepi-pc2.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h5-orangepi-pc2.dts
@@ -99,10 +99,6 @@ &codec {
 	status = "okay";
 };
 
-&cpu0 {
-	cpu-supply = <&reg_vdd_cpux>;
-};
-
 &de {
 	status = "okay";
 };
@@ -182,38 +178,10 @@ &ohci3 {
 	status = "okay";
 };
 
-&r_i2c {
-	status = "okay";
-
-	reg_vdd_cpux: regulator@65 {
-		compatible = "silergy,sy8106a";
-		reg = <0x65>;
-		regulator-name = "vdd-cpux";
-		silergy,fixed-microvolt = <1100000>;
-		regulator-min-microvolt = <1000000>;
-		regulator-max-microvolt = <1400000>;
-		regulator-ramp-delay = <200>;
-		regulator-boot-on;
-		regulator-always-on;
-	};
-};
-
 &sound_hdmi {
 	status = "okay";
 };
 
-&spi0  {
-	status = "okay";
-
-	flash@0 {
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "jedec,spi-nor";
-		reg = <0>;
-		spi-max-frequency = <40000000>;
-	};
-};
-
 &r_i2c {
 	status = "okay";
 
-- 
2.34.1

